sourcegraph/sourcegraph
# Cody App update notice+install experience

Implements the update notice in the Cody App home page and Admin Update page.

### Previous Behavior

When the Cody App launches, it checks for updates. If there's an update, it pops up a modal dialog and asks the user, once, if they want to upgrade. If the user declines, only restarting the application allows the popup to reappear.

This can be challenging to keep it up to date, as the Cody App does not usually quit but instead keeps running in the background to provide data to extensions [VScode, IntelliJ, etc.], and the user will never be notified an update exists unless they proactively close the app.

### New Behavior

Implements periodic checks in the background and user-controlled upgrade checks. The UI updates whenever there's an upgrade available. Allows for version check and installation directly from within the application at any time.

## Test plan

- Storybook for all visual components

- Functionality tested on Windows and MacOS

- Ensures the app can handle
  - No updates
  - Update available
  - Start install
  - Display errors if the download/install process failed

- Update strategies
  - Periodic
  - User check for updates

- Display information about current and new builds correctly

- Implemented Storybook stories for all states, state transitions, and UI placement.

### User Interface

Refer to Storybook @ cody-ui > Updater for visuals.
